dask/array/tiledb_io.py

```diff
@@ -27,17 +27,14 @@ def from_tiledb(uri, attribute=None, chunks=None, storage_options=None, **kwargs
     Examples
     --------
 
-    >>> # create a tiledb array
-    >>> import tiledb, numpy as np, tempfile  # doctest: +SKIP
-    >>> uri = tempfile.NamedTemporaryFile().name  # doctest: +SKIP
-    >>> tiledb.from_numpy(uri, np.arange(0,9).reshape(3,3))  # doctest: +SKIP
-    <tiledb.libtiledb.DenseArray object at 0x...>
-    >>> # read back the array
-    >>> import dask.array as da  # doctest: +SKIP
-    >>> tdb_ar = da.from_tiledb(uri)  # doctest: +SKIP
-    >>> tdb_ar.shape  # doctest: +SKIP
+    >>> import tempfile, tiledb
```

Comment:
I noticed this will require developers to have tiledb installed in their local software environments to run doctests.

Looking at the "Install" section of our contributing docs, people who use conda will have tiledb installed because they're using [this conda environment file](https://github.com/dask/dask/blob/8d49c3f1491b4aaccb6560814718f498abf63335/continuous_integration/environment-3.9.yaml) to set up their local environment. However, people using the `pip` approach won't have tiledb installed because the `complete` / `test` extras don't include tiledb (or lots of other optional dependencies). 

@jsignell @zzhengnan do you have any thoughts on how we want to handle this moving forward?

Reply:
A natural question that comes to mind is how come there's a divergence in the dependencies between what's specified in conda vs. pip? Feels to me the same dependencies should end up getting installed regardless of if a user decides to install via conda or pip.